### DISCUSSION TITLE
v1.10.0 feedback: disk polling support for `retype watch`

### DISCUSSION THREAD

**Discussion by fabriciomurta** [retypeapp/retype#151] [General]
We have implemented disk polling support to check for changes where `retype watch` can't "subscribe for file changes" and we're looking for feedback from the Retype community!

We are drafting the documentation for this feature [here](https://github.com/retypeapp/retype/blob/dev/configuration/project.md#watchpolling)

Basically:

```yaml
server:
  watch:
    polling: true|false|2000
```

Where `2000` above means "poll the input directory for changes (instead of subscribing) every 2 seconds". Default is `false` (use filesystem subscription).

:eyes: @DannyBen

**Comment by fabriciomurta** [comment]
By the way, in order to install the preview version, it is needed to explicitly specify it while installing:

## dotnet tool

#### New Install

```sh
dotnet tool install --global retypeapp --version 1.10.0-preview
```

#### Upgrade

```sh
dotnet tool update --global retypeapp --version 1.10.0-preview
```

#### Revert to stable

```sh
dotnet tool uninstall --global retypeapp
dotnet tool install --global retypeapp
```

#### Uninstall

```sh
dotnet tool uninstall --global retypeapp
```

## NPM

#### New install or upgrade

```sh
npm install --global retypeapp@1.10.0-preview
```

#### Revert to stable
```sh
npm install --global retypeapp
```

#### Uninstall
```sh
npm uninstall --global retypeapp
```

## yarn

#### New install
```sh
yarn global add retypeapp@1.10.0-preview
```

#### Upgrade
```sh
yarn global upgrade retypeapp@1.10.0-preview
```

#### Revert to stable
```sh
yarn global upgrade retypeapp@1.9.0
```

#### Uninstall
```sh
yarn global remove retypeapp
```

**Comment by fabriciomurta** [comment]
@DannyBen, one of the goals is that this feature allows you to use `retype watch` under your Vagrant environment. Have we attained it?

**Reply by DannyBen** [reply]
Wow - this is a very nice quality of life feature.

Now, my server section looks like this:

```yaml
server:
  host: 0.0.0.0
  port: 3000
  watch: { polling: 3000 }
```

and I have tested on my vagrant (which uses samba shared folder with the host) and it works beautifully.

I guess I can remove all my custom watch acrobatics now, which is very nice.

Thanks for following up, looking forward to the release.
Let me know if there is anything else that you want me to verify.

**Reply by fabriciomurta** [reply]
Thanks for the very very quick test and feedback, we are thrilled that the feature worked the way you needed!

**Reply by DannyBen** [reply]
After using this `retype watch` for a while, I can tell you this is a game changer for me.
As a Vagrant user, I do not have the luxury of live reload anywhere else, and with retype, this just works and the page is refreshed before I complete my Alt-Tab to my browser. 

So, thanks.
